sdk/servicebus/Microsoft.Azure.ServiceBus/src/Core/MessageSender.cs

```diff
@@ -573,7 +573,14 @@ async Task OnSendAsync(IList<Message> messageList)
                 }
                 catch (Exception exception)
                 {
-                    throw AmqpExceptionHelper.GetClientException(exception, amqpLink?.GetTrackingId(), null, amqpLink?.Session.IsClosing() ?? false);
+                    if (AmqpExceptionHelper.TryTranslateToRetriableException(exception, out var retriableEx))
+                    {
+                        throw retriableEx;
+                    }
+                    else
+                    {
+                        throw AmqpExceptionHelper.GetClientException(exception, amqpLink?.GetTrackingId(), null, amqpLink?.Session.IsClosing() ?? false);
```

Comment:
Neeraj once made a fix for this Github issue but not sure why it didn't resolve it and similar cases kept coming up: https://github.com/Azure/azure-sdk-for-net/pull/6940/files/1b6a6da7e95ecb0868691ae94948acaa27851b03
We have made sure their SDK had Neeraj's fix. Add the retriable exception to every place where this error might occur to attempt to resolve this incident.